rasa/nlu/tokenizers/tokenizer.py

```diff
@@ -105,6 +108,44 @@ def _split_intent(self, message: Message):
 
         return self._convert_words_to_tokens(words, text)
 
+    def _apply_token_pattern(self, tokens: List[Token]) -> List[Token]:
+        """Apply the token pattern to the given tokens.
+
+        Args:
+            tokens: list of tokens to split
+
+        Returns:
+            List of tokens.
+        """
+        if not self.token_pattern:
+            return tokens
+
+        token_pattern = re.compile(self.token_pattern)
```

Comment:
Let's move this to the constructor because it's a one time activity and also a costly operation.